### PR TITLE
Louisa mobile drawer UI fix

### LINKED ISSUE #2671: Mobile drawer UI fix
We would like to realign the content inside the Header and Drawer like thoses designs in [Figma](https://www.figma.com/file/LlopBqbeLND3IQQvYMjVbn/%F0%9F%9A%A7-Navigation-UX-%E2%80%93-Dawn-2023?type=design&node-id=2712%3A109199&t=ERNnDIdYRVGM6QG2-1).

- [ ] Burger/Close button
- [ ] Icons group (apps/cart/search)
- [ ] Menu items/Arrows
- [ ] Language
- [ ] Country/currency
- [ ]  Social icons
- [ ] Account icon/label

On mobile only.

### PR DESCRIPTION
**Pull request by lougoncharenko**
### PR Summary: 

Links, Social Icons, Language, Country/currency, and Account icon/label are aligned with close button and cart icon.

### Why are these changes introduced?

Fixes #2671 

### What approach did you take?
- Added vertical padding to the header
- Added vertical padding/ heading to the icons, Account icon/label, and Country/currency to line up with close button and cart icon

| # | Decision | Alternatives | Rationale | Downsides |
|---|---|---|---|---|
| 1 |   |   |   |   |

### Testing steps/scenarios
- [ ] Test the drawer in mobile setting on a phone
- [ ] Test mobile drawer in fullscreen - changes are for mobile only
- [ ] Test mobile drawer on desktop - changes are for mobile only
- [ ] Test color scheme and gradient to ensure nothing is broken 

### Demo links
- [Editor](https://os2-demo.myshopify.com/admin/themes/140057346070/editor?context=theme&category=gid%3A%2F%2Fshopify%2FOnlineStoreThemeSettingsCategory%2FSocial%2Bmedia%3Ftheme_id%3D140057346070%26first_setting_id%3Dsocial_facebook_link)

### Checklist
- [x] Added PR summary for [release notes](https://themes.shopify.com/themes/dawn/styles/default#ReleaseNotes)
- [x] Requested review from UX (Only for changes that are affecting the experience or perceivable visual details)
- [ ] Created a ticket for the [help.shopify.com](https://help.shopify.com) documentation team about updates to theme settings. (Internal-only task)
- [x] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [x] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [x] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [x] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [x] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

**Comment by lougoncharenko** [comment]
Screenshot of everything lining up
Screenshot 2023-06-06 at 11 27 16 AM

Video showing mobile drawer with mobile inspector
https://github.com/Shopify/dawn/assets/93098869/4a5baaad-172c-4e54-a9ba-d884515d7c7f

**Review by ludoboludo — COMMENTED** [review]
Looking good 👍 

Left a few comments

File: assets/component-localization-form.css [hunk 1]
```diff
@@ -283,7 +283,7 @@ noscript .localization-selector.link {
 .menu-drawer__localization .localization-form__select {
   background-color: initial;
   margin-top: 0;
-  padding: 1.5rem 3.6rem 1.5rem 1rem;
+  padding: 1rem 3.2rem 1rem 0rem;
```
[the remark this answers is not in the document]
Comment (lougoncharenko) [reply assets/component-localization-form.css:286]: I think I changed it to 3.2 rem by accident, thank you for calling it out! I am reverting it back to 3.6 rem for the right

File: assets/component-localization-form.css [hunk 2]
```diff
@@ -283,7 +283,7 @@ noscript .localization-selector.link {
...
 }
```
[the remark this answers is not in the document]
Comment (lougoncharenko) [reply assets/component-localization-form.css:287]: I removed the left margin and it still lines up with the other links and close button!

File: assets/component-menu-drawer.css [hunk 3]
```diff
@@ -223,23 +223,23 @@ details[open].menu-opening > .menu-drawer__submenu {
...
 .menu-drawer__account {
   display: inline-flex;
   align-items: center;
   text-decoration: none;
-  padding: 1.2rem;
+  padding: 1rem 0rem;
   font-size: 1.4rem;
   color: rgb(var(--color-foreground));
-  margin-bottom: 2rem;
+  margin-bottom: 0rem;
```
Comment (kmeleta) [inline assets/component-menu-drawer.css:237]: Nitpicks `padding: 1rem 0;` and `margin-bottom: 0s`

Comment (kmeleta) [reply assets/component-menu-drawer.css:237]: I just meant to remove the "rem" itself from the `0rem` values as its redundant and we typically just use `0` in these cases.

File: assets/component-menu-drawer.css [hunk 4]
```diff
@@ -202,7 +202,7 @@ details[open].menu-opening > .menu-drawer__submenu {
 
 .menu-drawer__close-button .icon-arrow {
   transform: rotate(180deg);
-  margin-right: 1rem;
+  margin-right: 3rem;
```
[the remark this answers is not in the document]
Comment (lougoncharenko) [reply assets/component-menu-drawer.css]: Sorry! I misunderstood , I made the suggestions as suggested! Thank you both :)

**Review by kmeleta — APPROVED** [review]
Perfect, thanks Louisa!

**Pull request merged**